### PR TITLE
Cleanup some effc++ warnings - September - part 2

### PR DESCRIPTION
**Pull request by dreamer**
Another boring review. This one clears out all *trivial* effc++ warnings, that were remaining (and brings us below 100 warnings with our current default set! :) ).

All remaining effc++ warnings are non-trivial or with an elevated risk of regressions - we'll need to fix those with extra care.

### CODE REVIEW

File: include/drives.h [hunk 1]
```diff
@@ -405,11 +405,18 @@ class Virtual_Drive: public DOS_Drive {
...
+	              const char *overlay,
+	              uint16_t _bytes_sector,
+	              uint8_t _sectors_cluster,
+	              uint16_t _total_clusters,
+	              uint16_t _free_clusters,
+	              uint8_t _mediaid,
+	              uint8_t &error);
 
-	virtual bool FileOpen(DOS_File * * file,char * name,Bit32u flags);
+	virtual bool FileOpen(DOS_File **file, char *name, Bit32u flags);
```
Comment (kcgen) [inline include/drives.h:419]: Bit32u flags) -> uint32_t

File: src/dos/drive_overlay.cpp [hunk 2]
```diff
@@ -304,9 +307,27 @@ static OverlayFile* ccc(DOS_File* file) {
 	return ret;
 }
 
-Overlay_Drive::Overlay_Drive(const char * startdir,const char* overlay, Bit16u _bytes_sector,Bit8u _sectors_cluster,Bit16u _total_clusters,Bit16u _free_clusters,Bit8u _mediaid,Bit8u &error)
-:localDrive(startdir,_bytes_sector,_sectors_cluster,_total_clusters,_free_clusters,_mediaid),special_prefix("DBOVERLAY") {
-	optimize_cache_v1 = true; //Try to not reread overlay files on deletes. Ideally drive_cache should be improved to handle deletes properly.
```
Comment (kcgen) [inline src/dos/drive_overlay.cpp:309]: Wow.. that's a truck load cleaned up! very nice.

File: src/hardware/dma.cpp [hunk 3]
```diff
@@ -264,22 +264,26 @@ Bitu DmaController::ReadControllerReg(Bitu reg,Bitu /*len*/) {
 	return 0xffffffff;
 }
 
-DmaChannel::DmaChannel(Bit8u num, bool dma16) {
-	masked = true;
-	callback = NULL;
-	if(num == 4) return;
-	channum = num;
```
Comment (kcgen) [inline src/hardware/dma.cpp:271]: If `num` is 4, then `channum` is left undefined?

**Review by kcgen — APPROVED** [review]
Looks great @dreamer! Thanks for cleaning up this remaining set of warnings. 
Tentatively approving, just one minor replacement and a question regarding the DMA `channum` initial value.

File: src/dos/drive_overlay.cpp (hunk 2 above)
Comment (dreamer) [reply src/dos/drive_overlay.cpp:309]: Yeah… someone had reaaaaaly wide monitor, I guess.

Funny thing - after I moved all members to be initialized in the initializer list, Clang detected `optimize_cache_v1` as unused variable (!). One more perk of following effc++ guidelines :)

File: src/hardware/dma.cpp (hunk 3 above)
Comment (dreamer) [reply src/hardware/dma.cpp]: Great find, thank you!

This `if` statement generally looks fishy, but I don't want to jump into another rabbit hole just yet ;)

**Comment by kcgen** [comment]
Looks good @dreamer!

**Comment by dreamer** [comment]
Thanks @kcgen :)

**Pull request merged**